### PR TITLE
PR only containing CMakefiles

### PR DESCRIPTION
**Pull request by Itay-Lifshitz**
As we agreed, I am opening 2 PRs separating source code changes and CMakefiles changes.
But you should know that it fails to compile on Windows platform if another PR is not merged.

### CODE REVIEW

**Comment by Itay-Lifshitz** [comment]
Based on your last comment, I guess you didn't install `cmake` and `ming32-make` in MSYS2.
Please read the instructions in qmake2cmake.md carefully.
```bash
pacman -S mingw32/mingw-w64-i686-cmake mingw32/mingw-w64-i686-make
```

**Comment by mcallegari** [comment]
> Based on your last comment, I guess you didn't install `cmake` and `ming32-make` in MSYS2. Please read the instructions in qmake2cmake.md carefully.
> 
> ```shell
> pacman -S mingw32/mingw-w64-i686-cmake mingw32/mingw-w64-i686-make
> ```

You guess? Then you probaby didn't open the log I attached, which shows that cmake and make are actually installed in the system.
Here's the log again.
[msys2_build_fail.txt](https://github.com/mcallegari/qlcplus/files/11966148/msys2_build_fail.txt)

**Comment by Itay-Lifshitz** [comment]
@mcallegari I actually already checked your attachment and found the following section.
`
massimo@DESKTOP-7VPVK4K MINGW32 /c/projects/qlcplus/build
$ cmake -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=Release ..
CMake Error: Could not create named generator MinGW Makefiles
`
It means that you did not install `ming32-make` in MSYS2. 
*ATTENTION*: it is not `make`, but `ming32-make` instead.
Or are you sure that you ran these commands in MSYS2 mingw32 shell instead of normal command prompt?

**Comment by mcallegari** [comment]
OK now I get the difference. I always thought mingw32-make and make were aliases of the same package, but apparently they're not.
Curious thing is that plain "make" doesn't require patches to cpp files, so I'm wondering what is the difference and if it is possible to use cmake+make without build errors.
Any clue?

**Comment by Itay-Lifshitz** [comment]
@mcallegari We actually can compile without ming32-make.
I uninstalled `mingw32-make` and tried to compile using the following commands, and it worked.
```
cmake -G "Unix Makefiles" ..
make
```
But it still required me to merge cmake_cpp changes, otherwise it would throw the following errors.
```
[  0%] Automatic MOC and UIC for target dmxusb
[  0%] Built target dmxusb_autogen
[  0%] Building CXX object plugins/dmxusb/src/CMakeFiles/dmxusb.dir/dmxusb_autogen/mocs_compilation.cpp.obj
In file included from D:/msys64/mingw32/include/oleidl.h:7,
                 from D:/msys64/mingw32/include/ole2.h:38,
                 from D:/msys64/mingw32/include/wtypes.h:13,
                 from D:/msys64/mingw32/include/winscard.h:10,
                 from D:/msys64/mingw32/include/windows.h:97,
                 from E:/Projects/C/qlcplus/plugins/interfaces/qlcioplugin.h:35,
                 from E:/Projects/C/qlcplus/plugins/dmxusb/src/dmxusb.h:24,
                 from E:/Projects/C/qlcplus/build/plugins/dmxusb/src/dmxusb_autogen/EWIEGA46WW/moc_dmxusb.cpp:10,
                 from E:/Projects/C/qlcplus/build/plugins/dmxusb/src/dmxusb_autogen/mocs_compilation.cpp:2:
E:/Projects/C/qlcplus/plugins/dmxusb/src/dmxusbwidget.h:62:32: error: expected ',' or '...' before 'struct'
   62 |     DMXUSBWidget(DMXInterface *interface, quint32 outputLine, int frequency);
      |                                ^~~~~~~~~
E:/Projects/C/qlcplus/plugins/dmxusb/src/dmxusbwidget.h:90:19: error: expected unqualified-id before 'struct'
   90 |     DMXInterface *interface() const;
      |                   ^~~~~~~~~
E:/Projects/C/qlcplus/plugins/dmxusb/src/dmxusbopenrx.h:42:32: error: expected ',' or '...' before 'struct'
   42 |     DMXUSBOpenRx(DMXInterface *interface,
      |                                ^~~~~~~~~
E:/Projects/C/qlcplus/plugins/dmxusb/src/enttecdmxusbopen.h:46:36: error: expected ',' or '...' before 'struct'
   46 |     EnttecDMXUSBOpen(DMXInterface *interface,
      |                                    ^~~~~~~~~
E:/Projects/C/qlcplus/plugins/dmxusb/src/enttecdmxusbpro.h:72:40: error: expected ',' or '...' before 'struct'
   72 |     EnttecDMXUSBProInput(DMXInterface *interface);
      |                                        ^~~~~~~~~
E:/Projects/C/qlcplus/plugins/dmxusb/src/enttecdmxusbpro.h:101:35: error: expected ',' or '...' before 'struct'
  101 |     EnttecDMXUSBPro(DMXInterface *interface,
      |                                   ^~~~~~~~~
make[2]: *** [plugins/dmxusb/src/CMakeFiles/dmxusb.dir/build.make:117: plugins/dmxusb/src/CMakeFiles/dmxusb.dir/dmxusb_autogen/mocs_compilation.cpp.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:5457: plugins/dmxusb/src/CMakeFiles/dmxusb.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
I tried to use gcc and g++ instead of default cmake compilers like cc and c++, but it did not work.

**Comment by mcallegari** [comment]
Just a quick update. I've been able to make the first build on Windows
I pushed a couple of changes on cmakebuild branch (please rebase)
Now installation on Windows is completely messed up:
- all qm files should be packed into a single one as it was with qmake
- libqlcpluswebaccess.dll goes into a bin folder
- .a files shouldn't be installed at all
- all Qt libraries are missing
- all system libraries are missing
- fixtures are not separated into manufacturer folders
- icons are missing on exe files
- NSIS file is missing

**Comment by Itay-Lifshitz** [comment]
@mcallegari OK, let me check.
But can you please check if the current qmake works on Windows?
I tried to install on Windows using qmake but it did not install dependent Qt and system libraries.

File: engine/test/qlci18n/.gitignore
Comment (mcallegari) [inline engine/test/qlci18n/.gitignore:1]: is this a mistake?

File: engine/test/qlci18n/qlci18n_fi_FI.ts
Comment (mcallegari) [inline engine/test/qlci18n/qlci18n_fi_FI.ts:1]: I pushed this change. Please update your fork

File: plugins/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 project(plugins)
 
+# libusb-1.0
+pkg_check_modules(LIBUSB_1 IMPORTED_TARGET libusb-1.0)
```
Comment (mcallegari) [inline plugins/CMakeLists.txt:4]: why is this here?
It doesn't belong to every plugin

File: plugins/CMakeLists.txt [hunk 2]
```diff
@@ -23,7 +26,7 @@ if((((NOT ANDROID AND NOT IOS) AND (UNIX) AND (${LIBOLA_FOUND}) AND (${LIBOLASER
 endif()
 if(UNIX AND NOT ANDROID AND NOT IOS AND NOT MACOS)
     add_subdirectory(dmx4linux)
+    add_subdirectory(spi)
```
Comment (mcallegari) [inline plugins/CMakeLists.txt:29]: why have you re-enabled it?
SPI on desktop doesn't make any sense.
Please leave as upstream version

File: plugins/dmxusb/src/CMakeLists.txt
Comment (mcallegari) [inline plugins/dmxusb/src/CMakeLists.txt:1]: please leave as upstream
I'll take care of platforms

File: ui/src/CMakeLists.txt [hunk 3]
```diff
@@ -20,7 +20,6 @@ else()
 endif()
 
 if((((QT_VERSION_MAJOR LESS 5)) AND (MACOS)))
```
Comment (mcallegari) [inline ui/src/CMakeLists.txt:22]: Qt4 is no longer supported
In case, leave portaudio stuff commented somewhere but I doubt I will pick it up again

File: engine/src/qlcconfig.h.unix.in
Comment (mcallegari) [inline engine/src/qlcconfig.h.unix.in:1]: Why two .in files?? unix and win32 versions are identical.
Let's keep a single one as it was with qmake

**Comment by mcallegari** [comment]
Please see my comments.
It's very difficult to review this PR since it includes different topics.
Let's take care of each topic one by one. It's faster for everyone.
Please fix installation first, then we can see the rest

File: engine/src/qlcconfig.h.unix.in
Comment (Itay-Lifshitz) [reply engine/src/qlcconfig.h.unix.in:1]: It is my fault. They were different and they should be different, but the content of the qlcconfig.h.win32.in file has been changed during the auto rebase process.
I will commit the different one.

File: engine/test/qlci18n/.gitignore
Comment (Itay-Lifshitz) [reply engine/test/qlci18n/.gitignore:1]: Not a mistake. I deleted it by hands, because it included qlci18n_fi_FI.ts which is required by project file and should be managed by git.

File: plugins/CMakeLists.txt (hunk 1 above)
Comment (Itay-Lifshitz) [reply plugins/CMakeLists.txt:4]: You are correct, but it is used in at least 2 directories and I did not want to check modules twice.

File: plugins/CMakeLists.txt (hunk 2 above)
Comment (Itay-Lifshitz) [reply plugins/CMakeLists.txt:29]: OK, I enabled it because spi is enabled in QT .pro file.

**Comment by Itay-Lifshitz** [comment]
@mcallegari 
I think we need to sort out the workflow on the PR first of all.
I open the PR and commit changes on my "cmake_files" branch.
You review it and merge it on your "cmakebuild" branch manually, so you make commits on your branch.
And you require me to rebase to your branch and make fixes.
But when I try to rebase my local branch to your remote branch like "git checkout cmake_files && git rebase upstream/cmakebuild", it always conflicts because you already make changes on your side manually.
It is too annoying for me to resolve the conflicts all the time.
So let's discuss this workflow first.

**Comment by mcallegari** [comment]
@Itay-Lifshitz as I mentioned before, it's quite difficult for me too to handle a PR with dozens of files which includes several topics, mistakes, opinions, code changes, etc
My goal is to progress on this so I'm trying to dedicate some of my time (that I could otherwise dedicate to something else in this project) to make little steps but steady.
The alternative is that I review and reject your PR forever, doing 0% progress.

I have update #1408 with the task list. As you can see, we barely covered 2 items because you haven't tested your changes by comparing them to the qmake build.
In my opinion the workflow should address each one of those items at the time. Not everything all together messed up.
So let's concentrate on this now: `Qt5 v4 deploy on Windows (.exe bundle)`

What is missing upstream:
- exe icons (you already made this chnge but I haven't ported it)
- translations installation (now they're packed together but not installed)
- test NSIS script (I can do this)

Can you please send a PR only for this? Thanks

**Comment by Itay-Lifshitz** [comment]
OK, let me close this PR and open one PR for each task in the checklist of #1408 .
